CHANGELOG.md

```diff
@@ -32,6 +56,107 @@ Now we'll checking for links in Pending state, and if they haven't hit a configu
 
+## Enhanced CLI Login Functionality
+### Server Trust
+#### Untrusted Servers
+If you don't provide a certificates file when logging in, the server's well known certificates will now be pulled from the server and you will be prompted if you want to use them. 
+If certs for the host have previously been retrieved they will be used. Certs stored locally will be checked against the certs on the server when logging in. 
+If a difference is found, the user will be notified and asked if they want to update the local certificate cache.
+
+If you provide certificates during login, the server's certificates will not be checked or downloaded. Locally cached certificates for that host will not be used.
+
+#### Trusted Servers
+If working with a server which is using certs that your OS already recognizes, nothing will change. No cert needs to be provided and the server's well known certs will not be downloaded.
+
+### Identities
+The Ziti CLI now suports multiple identities. An identity can be specified using `--cli-identity` or `-i`. 
+
+Example commands:
+```
+$ ziti edge login -i dev localhost:1280
+Enter username: admin
+Enter password: 
+Token: 76ff81b4-b528-4e2c-ad73-dcb0a39b6489
+Saving identity 'dev' to ~/.config/ziti/ziti-cli.json
+
+$ ziti edge -i dev list services
+id: -JucPW0kGR    name: ssh    encryption required: true    terminator strategy: smartrouting    role attributes: ["ssh"]
+results: 1-1 of 1
+```
+
+If no identity is specified, a default will be used. The default identity is `default`. 
+
+#### Switching Default Identity
+The default identity can be changed with the `ziti edge use` command.
+
+The above example could also be accomplished as follows:
+```
+$ ziti edge use dev
+Settting identity 'dev' as default in ~/.config/ziti/ziti-cli.json
+
+$ ziti edge login localhost:1280
+Enter username: admin
+Enter password: 
+Token: e325d91c-a452-4454-a733-cfad88bfa356
+Saving identity 'dev' to ~/.config/ziti/ziti-cli.json
+
+$ ziti edge list services
+id: -JucPW0kGR    name: ssh    encryption required: true    terminator strategy: smartrouting    role attributes: ["ssh"]
+results: 1-1 of 1
+
+$ ziti edge use default
+Settting identity 'default' as default in ~/.config/ziti/ziti-cli.json
+```
+
+`ziti edge use` without an argument will list logins you have made.
+
+```
+$ ziti edge use
+id:      default | current:  true | read-only:  true | urL: https://localhost:1280/edge/management/v1
+id:        cust1 | current: false | read-only: false | urL: https://customer1.com:443/edge/management/v1
+```
+
+#### Logout
+You can now also clear locally stored credentials using `ziti edge logout`
+
+```
+$ ziti edge -i cust1 logout  
+Removing identity 'cust1' from ~/.config/ziti/ziti-cli.json
+```
+
+#### Read-Only Mode
+When logging in one can mark the identity as read-only. This is a client side enforced flag which will attempt to make sure only
+read operations are performed by this session. 
+
+```
+$ ziti edge login --read-only localhost:1280
```

Comment:
should read-only be the default? force people to write mode?

Reply:
I think most people probably want it in RW mode. Especially for new developers who are trying out the system I'd hate to put a roadblock in their way as they are experimenting. 

One thing that might make sense is it use read-only by default if you are using --token, since that implies a workflow where you've been granted temporary access to a system.

Thoughts?